File: iOS/XXXX/XXXX/ViewController.m
//
//  ViewController.m
//  XXXX
//
//  Created by le on 2019/5/8.
//  Copyright © 2019 le. All rights reserved.
//

#import "ViewController.h"

typedef void(^BoringBlock)(void);

@interface ViewController ()
{
    NSString *_string;
    BoringBlock _blockhead;
}
@end

@implementation ViewController

static const struct block_descriptor_NAME {
    unsigned long reserved;
    unsigned long literal_size;
    void (*copy_helper)(void *dst,void *src);
    void (*dispose_helper)(void *src);
};

struct block_literal_NAME {
    void *isa;
    int flags;
    int reserved;
    void (*invoke)(void *literal,...);
    struct block_descriptor_NAME *descriptro;
};

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
//    _string = @"snork";
//    _blockhead = Block_copy(^{
//        NSLog(@"string is %@",self->_string);
//    });
    
}


@end

File: DataStructures&Algorithms/DataLearn/DataLearn/ViewController.swift
//
//  ViewController.swift
//  DataLearn
//
//  Created by le on 2018/7/20.
//  Copyright © 2018年 le. All rights reserved.
//

import UIKit


class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        let list = LinkedList<String>()
        list.isEmpty
        list.first
        list.appendLinkList(value: "hello")
        list.appendLinkList(value: "sb")
        
        let Node = list.node(atIndex: 0)
        print(Node.value)
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}



